# 📊 Paytm Financial Health Analysis
## Deep Dive into India's Leading Fintech Company

**Analyst:** Swetarani Sa
**Date:** January 17, 2026  
**Company:** One 97 Communications (Paytm)  
**Ticker:** PAYTM.NS

---

### Executive Summary

This analysis examines Paytm's financial health following the RBI's payment bank ban in February 2024. Using publicly available financial data, I assess:

- **Leverage** (Debt burden)
- **Profitability** (Are they making money?)
- **Liquidity** (Can they pay bills?)
- **Market Perception** (Investor confidence)

**Goal:** Determine if Paytm is a viable investment/lending opportunity in 2026.

---

In [24]:
#importing necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import yfinance as yf
import warnings 
warnings.filterwarnings('ignore')

In [25]:
# Fetching data
paytm = yf.Ticker("PAYTM.NS")
info = paytm.info
info

{'address1': 'One Skymark, Tower-D',
 'address2': 'Floor No. 6th to 22nd Plot No. H-10B Sector 98',
 'city': 'Noida',
 'zip': '201304',
 'country': 'India',
 'phone': '91 12 0477 0770',
 'fax': '91 12 0477 0771',
 'website': 'https://www.paytm.com',
 'industry': 'Software - Infrastructure',
 'industryKey': 'software-infrastructure',
 'industryDisp': 'Software - Infrastructure',
 'sector': 'Technology',
 'sectorKey': 'technology',
 'sectorDisp': 'Technology',
 'longBusinessSummary': 'One97 Communications Limited provides payment, commerce and cloud, and financial services to consumers and merchants in India, the United Arab Emirates, Saudi Arabia, and Singapore. The company offers payment and financial services, including payment facilitator services, facilitation of consumer and merchant lending to consumers and merchants, wealth management, etc. It also provides marketing services consist of aggregators for digital products, the ticketing business, the provision of voice and messaging

In [26]:
# Fetching Basic Data
print("PAYTM Information:")
print(f"Company Name: {info.get('longName', 'N/A')}")
print(f"Sector: {info.get('sector', 'N/A')}")
print(f"Industry: {info.get('industry', 'N/A')}")
print(f"Market Cap: {info.get('marketCap', 'N/A')}")
print(f"Current Price: {info.get('currentPrice', 'N/A')}")
print(f"52 Week High: {info.get('fiftyTwoWeekHigh','N/A')}")
print(f"52 Week Low: {info.get('fiftyTwoWeekLow','N/A')}")
print(f"52 Week Change: {info.get('52WeekChange','N/A')}")

PAYTM Information:
Company Name: One97 Communications Limited
Sector: Technology
Industry: Software - Infrastructure
Market Cap: 858834993152
Current Price: 1342.5
52 Week High: 1381.8
52 Week Low: 651.5
52 Week Change: 0.49399066


### Initiqal Observation
- Current Price : ₹1288.6
- Down 40.1% from it's IPO price of ₹2150
- 52 week change shows 63% rise, indicating high volatility
Despite it's 63% recovery over the past year, the stock is still about 40% below it's IPO price, indicating high volatility.

In [27]:
balance_sheet = paytm.balance_sheet
if balance_sheet.empty:
    print("No balance sheet data available.")
else:
    latest_bs = balance_sheet.iloc[:, 0]
    
    # Extracting key balance sheet metrics
    total_asset = latest_bs.get('Total Assets', 0)
    total_liability = latest_bs.get('Total Liabilities Net Minority Interest', 0)
    current_assets = latest_bs.get('Current Assets', 0)
    current_liabilities = latest_bs.get('Current Liabilities', 0)
    total_debt = latest_bs.get('Total Debt', 0)
    equity = latest_bs.get('Total Equity Gross Minority Interest', 0)
    minority_interest = latest_bs.get('Minority Interest', 0)


    # Displaying Balance Sheet Highlights
    print("\nBalance Sheet Highlights:")
    print(f"\nTotal Assets: ₹{total_asset/1e7} Crores")
    print(f"\nCurrent Assets: ₹{current_assets/1e7} Crores")
    print(f"\nTotal Liabilities: ₹{total_liability/1e7} Crores")
    print(f"\nCurrent Liabilities: ₹{current_liabilities/1e7} Crores")
    print(f"\nTotal Debt: ₹{total_debt/1e7} Crores")
    print(f"\nTotal Stockholder Equity: ₹{equity/1e7} Crores")
    print(f"\nMinority Interest: ₹{minority_interest/1e7} Crores")

bs_metrics = {
    'Total Assets': total_asset,
    'Current Assets': current_assets,
    'Total Liabilities': total_liability,
    'Current Liabilities': current_liabilities,
    'Total Debt': total_debt,
    'Total Stockholder Equity': equity  
}


Balance Sheet Highlights:

Total Assets: ₹21447.7 Crores

Current Assets: ₹17084.5 Crores

Total Liabilities: ₹6450.6 Crores

Current Liabilities: ₹6186.4 Crores

Total Debt: ₹160.4 Crores

Total Stockholder Equity: ₹14997.1 Crores

Minority Interest: ₹-29.6 Crores


### Insights
- The company has a strong liquidity position with current assets being almost 3 times of current liabilities,indication low short term default risk.
- The total debt is very small compared to both total assets and equity, which is a good sign as the company does not run on borrowed money.
- Equity finance is creating the majority of assests here.
- Overall, the balance sheet seems financially stable , with risk more likely arising from operations rather than solvency.

In [28]:
### GETTING INCOME STATEMENT DATA
income_statement = paytm.income_stmt
if income_statement.empty:
    print("No income statement data is available.")
else:
    latest_is = income_statement.iloc[:, 0]
    
# Extracting key income statement metrics
    total_revenue = latest_is.get('Total Revenue', 0) 
    total_expenses = latest_is.get('Total Expenses',0)
    net_income = latest_is.get('Net Income',0)
    ebit = latest_is.get('EBIT',0)
    ebitda = latest_is.get('EBITDA',0)
    operating_income = latest_is.get('Operating Income',0)
    cost_of_revenue = latest_is.get('Cost Of Revenue',0)
    gross_profit = latest_is.get('Gross Profit',0)
    interest_expense = latest_is.get('Interest Expense',0)

    # Displaying Income Statement Highlights
    print("\nIncome Statement Highlights:")
    print(f"\nTotal Revenue: ₹{total_revenue/1e7} Crores")
    print(f"\nTotal Expenses: ₹{total_expenses/1e7} Crores")
    print(f"\nNet Income: ₹{net_income/1e7} Crores")  
    print(f"\nEBIT: ₹{ebit/1e7} Crores")
    print(f"\nEBITDA: ₹{ebitda/1e7} Crores")
    print(f"\nOperating Income: ₹{operating_income/1e7} Crores")
    print(f"\nCost of Revenue: ₹{cost_of_revenue/1e7} Crores")
    print(f"\nGross Profit: ₹{gross_profit/1e7} Crores")
    print(f"\nInterest Expense: ₹{interest_expense/1e7} Crores")

is_metrics = {
    'Total Revenue': total_revenue,
    'Total Expenses': total_expenses,
    'Net Income': net_income,
    'EBITDA': ebitda,
    'Operating Income': operating_income,
    'Cost of Revenue': cost_of_revenue,
    'Gross Profit': gross_profit,
    'Interest Expense': interest_expense
}


Income Statement Highlights:

Total Revenue: ₹6740.8 Crores

Total Expenses: ₹8543.8 Crores

Net Income: ₹-658.7 Crores

EBIT: ₹-628.8 Crores

EBITDA: ₹43.8 Crores

Operating Income: ₹-1803.0 Crores

Cost of Revenue: ₹5123.9 Crores

Gross Profit: ₹1616.9 Crores

Interest Expense: ₹16.4 Crores


In [29]:
cash_flow = paytm.cashflow
if cash_flow.empty:
    print("No cash flow data available.")
else:
    latest_cf = cash_flow.iloc[:,0]
    # print(latest_cf)
    # Extracting key cash flow metrics
    operating_cash_flow =  latest_cf.get('Operating Cash Flow',0)
    free_cash_flow = latest_cf.get('Free Cash Flow',0)
    capital_expenditures = latest_cf.get('Capital Expenditure',0)
    beginning_cash_position = latest_cf.get('Changes In Cash',0) 
    # Displaying Cash Flow Highlights
    print("\nCash Flow Highlights:")
    print(f"\nOperating Cash Flow: ₹{operating_cash_flow/1e7} Crores")
    print(f"\nFree Cash Flow: ₹{free_cash_flow/1e7} Crores")
    print(f"\nCapital Expenditures: ₹{capital_expenditures/1e7} Crores")
    print(f"\nChange In Cash: ₹{beginning_cash_position/1e7} Crores")

cf_metrics = {
    'Operating Cash Flow': operating_cash_flow,
    'Free Cash Flow': free_cash_flow,
    'Capital Expenditures': capital_expenditures,
    'Change In Cash': beginning_cash_position
}


Cash Flow Highlights:

Operating Cash Flow: ₹-121.3 Crores

Free Cash Flow: ₹-443.2 Crores

Capital Expenditures: ₹-321.9 Crores

Change In Cash: ₹-2216.9 Crores


In [30]:
### LEVERAGE METRICS
### DEBT TO EQUITY RATIO
print("\nDebt-to-Equity Ratio Analysis:")
if equity != 0:
    debt_to_equity_ratio = total_debt / equity
    print(f"Debt to Equity Ratio: {debt_to_equity_ratio:.2f}")
    if debt_to_equity_ratio < 0.5:
        print("The company has a very low debt to equity ratio, indicating very low financial risk.")
    elif debt_to_equity_ratio < 1:
        print("The company has a low debt to equity ratio, indicating lower financial risk.")
    elif debt_to_equity_ratio < 2:
        print("The company has a moderate debt to equity ratio.")
    else:
        print("The company has a high debt to equity ratio, indicating higher financial risk.")
    bs_metrics['debt_to_equity'] = debt_to_equity_ratio




Debt-to-Equity Ratio Analysis:
Debt to Equity Ratio: 0.01
The company has a very low debt to equity ratio, indicating very low financial risk.


In [31]:
### INTEREST COVERAGE RATIO
print("\nInterest Coverage Ratio Analysis:")
if interest_expense != 0:
    interest_coverage_ratio = operating_income / interest_expense
    print(f"Interest Coverage Ratio: {interest_coverage_ratio:.2f}")
    if interest_coverage_ratio > 5:
        print("The company has a strong ability to cover its interest expenses.")
    elif interest_coverage_ratio > 2:
        print("The company has a moderate ability to cover its interest expenses.")
    else:
        print("The company may struggle to cover its interest expenses.")

    bs_metrics['interest_coverage'] = interest_coverage_ratio


Interest Coverage Ratio Analysis:
Interest Coverage Ratio: -109.94
The company may struggle to cover its interest expenses.


In [32]:
### PROFITABILITY METRICS
### Return on Equity (ROE)
print("\nReturn on Equity (ROE):")
if equity != 0:
    roe = net_income/equity
    print(f"Return on Equity: {roe*100:.2f}%")
    if roe > 0.15:
        print("The company has a strong return on equity, indicating efficient use of equity capital.")
    elif roe > 0.05:
        print("The company has a moderate return on equity.")
    elif roe > 0:
        print("The company has a low return on equity.")
    else:
        print("The company has a negative return on equity, indicating losses.")

    is_metrics['ROE'] = roe


Return on Equity (ROE):
Return on Equity: -4.39%
The company has a negative return on equity, indicating losses.


In [33]:
### Operating Margin
print("\nOperating Margin Analysis:")
if total_revenue !=0:
    operating_margin = operating_income/ total_revenue
    print(f"Operating Margin: {operating_margin*100:.2f}%")
    if operating_margin > 0.2:
        print("The company has a strong operating margin, indicating efficient operations.")
    elif operating_margin > 0.1:
        print("The company has a moderate operating margin.")
    elif operating_margin > 0:
        print("The company has a low operating margin.")
    else:
        print("The company has a negative operating margin, indicating operational losses.")
    
is_metrics['operating_margin'] = operating_margin


Operating Margin Analysis:
Operating Margin: -26.75%
The company has a negative operating margin, indicating operational losses.


In [34]:
### LIQUIDITY METRICS
### Current Ratio
print("\nCurrent Ratio Analysis:")
if current_liabilities !=0:
    current_ratio = current_assets/current_liabilities
    print(f"Current Ratio: {current_ratio:.2f}")
    if current_ratio > 2.0:
        print("The company has a strong current margin, indicating healthy liquidity.")
    elif current_ratio >1.5:
        print("The company has a good current ratio, indicating good liquidity.")
    elif current_ratio > 1.0:
        print("The company has a moderate current ratio, indicating adequate liquidity.")
    else:
        print("The company has very bad liquidity.")




Current Ratio Analysis:
Current Ratio: 2.76
The company has a strong current margin, indicating healthy liquidity.


In [35]:
### STOCK VOLATILITY AND MARKET PERFORMANCE ANALYSIS
print("\nStock Volatility and Market Performance Analysis:")
hist = paytm.history(period="1y")
if not hist.empty:
    start_price = hist['Close'].iloc[0]
    end_price = hist['Close'].iloc[-1]
    price_change = ((end_price - start_price) / start_price) * 100
    returns = hist['Close'].pct_change().dropna()
    volatility = returns.std()*(252**0.5)*100
print(f"Stock Price 1 Year Ago: ₹{start_price:.2f}")
print(f"Current Stock Price: ₹{end_price:.2f}")
print(f"Price Change over 1 Year: {price_change:.2f}%")
print(f"Stock Price Volatility: {volatility:.2f}%")

print("\n📊 Benchmark Comparison:")
if volatility< 30:
    print("The stock is a stable,blue-chip stock with low volatility.")
elif volatility < 50:
    print("The stock is a growth stock with moderate volatility.")
elif volatility < 80:
    print("The stock is a high-risk stock with significant volatility.")
else:
    print("The stock is an extremely risky stock with very high volatility.")

bs_metrics[volatility]= volatility
bs_metrics[price_change]= price_change


Stock Volatility and Market Performance Analysis:


Stock Price 1 Year Ago: ₹895.00
Current Stock Price: ₹1342.50
Price Change over 1 Year: 50.00%
Stock Price Volatility: 37.22%

📊 Benchmark Comparison:
The stock is a growth stock with moderate volatility.


In [36]:
### OVERALL HEALTH CHECK SUMMARY
print("\nOverall Health Check Summary:")
heath_score = 0
heath_factors = []
positive_factors = []

if debt_to_equity_ratio < 1.0:
    heath_score += 2
    heath_factors.append("Low Debt to Equity Ratio")
    positive_factors.append("Low Debt to Equity Ratio")
if interest_coverage_ratio > 2:
    heath_score += 2
    heath_factors.append("Strong Interest Coverage Ratio")
    positive_factors.append("Strong Interest Coverage Ratio")
if roe > 0.1:
    heath_score += 2
    heath_factors.append("Healthy Return on Equity")
    positive_factors.append("Healthy Return on Equity")
if operating_margin > 0.1:
    heath_score += 2
    heath_factors.append("Good Operating Margin")
    positive_factors.append("Good Operating Margin")
if current_ratio > 1.5:
    heath_score += 2
    heath_factors.append("Strong Current Ratio")
    positive_factors.append("Strong Current Ratio")

    
print(f"\nHealth Score: {heath_score}/10")

print("\nPositive Factors:")
for factor in positive_factors:
    print(f" - {factor}")

print("\nAreas of Concern:")
for factor in heath_factors:
    print(f" - {factor}")

if heath_score >= 8:
    print("\nOverall Assessment: The company is in excellent financial health.")    
elif heath_score >= 5:
    print("\nOverall Assessment: The company is in good financial health with some areas to monitor.")
elif heath_score >= 3:
    print("\nOverall Assessment: The company has moderate financial health and should address certain issues.")
else:
    print("\nOverall Assessment: The company is in poor financial health and requires immediate attention to its financial practices.")



Overall Health Check Summary:

Health Score: 4/10

Positive Factors:
 - Low Debt to Equity Ratio
 - Strong Current Ratio

Areas of Concern:
 - Low Debt to Equity Ratio
 - Strong Current Ratio

Overall Assessment: The company has moderate financial health and should address certain issues.


## 📋 Executive Summary: The Paytm Paradox

**Date:** January 19, 2026  
**Company:** One97 Communications Limited (Paytm)  
**Overall Assessment:** MODERATE-TO-HIGH RISK (4/10 Health Score)

---

### The Fundamental Contradiction

Paytm presents a striking paradox: a company that appears financially secure on paper yet faces serious questions about long-term viability. Strong liquidity and minimal debt coexist with persistent operational losses and cash burn.

---

### Key Findings

**Financial Strengths:**
- ✅ **Excellent Liquidity** - Current Ratio of 2.76 indicates strong ability to meet short-term obligations
- ✅ **Minimal Leverage** - D/E of 0.01 means virtually no debt burden
- ✅ **Substantial Asset Base** - ₹21,448 Cr in total assets, with ₹17,085 Cr readily accessible
- ✅ **Stock Recovery** - Up 50% over the past year, showing some market confidence

**Critical Weaknesses:**
- 🚨 **Core Business Unprofitable** - Operating Margin of -26.75% means every ₹100 of revenue loses ₹26.75
- 🚨 **Shareholder Value Destruction** - ROE of -4.39% indicates losses on equity investment
- 🚨 **Severe Cash Burn** - Quarterly cash decline of ₹2,217 Cr suggests unsustainable burn rate
- 🚨 **Operational Inefficiency** - Operating loss of ₹1,803 Cr reveals fundamental business model challenges
- 🚨 **Negative Cash Generation** - Free Cash Flow of ₹-443 Cr shows business cannot self-fund

---

### The Central Issue: Time vs. Viability

**The Core Problem:**
After 14 years of operation, Paytm has yet to prove its business model can generate sustainable profits. The company is not facing imminent bankruptcy, but it's in a race against time.

**Runway Analysis:**
- Current quarterly cash burn: ~₹2,217 Cr
- Annualized burn rate: ~₹8,868 Cr/year
- Current assets available: ₹17,085 Cr
- **Estimated runway:** Approximately 2-3 years before requiring additional capital

**What the IPO Money is Hiding:**
The strong balance sheet (₹15,000 Cr equity) is primarily from:
- ₹18,300 Cr raised in 2021 IPO
- Prior investor funding rounds
- **NOT** from profitable operations

This masks a fundamental truth: the business itself consumes cash rather than generates it.

---

### Profitability Deep Dive

**Revenue vs. Reality:**
- Total Revenue: ₹6,741 Cr (respectable scale)
- Total Expenses: ₹8,544 Cr
- **Gap:** ₹1,803 Cr operating loss

**To reach breakeven, Paytm must:**
1. Increase revenue by 27% (while holding costs constant) - **unlikely given competition**
2. Cut expenses by 21% (₹1,803 Cr) - **would require massive restructuring**
3. Achieve some combination of both - **most realistic but challenging**

**The EBITDA Illusion:**
- EBITDA: ₹44 Cr (barely positive at 0.65% margin)
- EBIT: ₹-629 Cr (deeply negative after real costs)

When management highlights "EBITDA positive," recognize this as the most generous possible accounting. Even excluding depreciation, amortization, interest, and taxes, they're essentially at breakeven on ₹6,741 Cr of revenue.

---

### The Debt Paradox

**Why Near-Zero Debt is Concerning:**
Most analysts see D/E of 0.01 as positive. I see it as a red flag.

**Counter-intuitive insight:**
Paytm has minimal debt NOT because they're financially disciplined, but likely because:
1. **Banks won't lend** - Post-RBI ban, credit markets are skeptical
2. **Too risky for traditional debt** - Negative cash flow makes them unlendable
3. **Forced to dilute equity** - Only funding option available

**Evidence:**
- Interest expense of only ₹16.4 Cr on ₹21,448 Cr in assets is abnormally low
- Healthy companies CHOOSE not to borrow; distressed companies CAN'T borrow
- Paytm likely falls in the latter category

---

### Cash Flow: The Unvarnished Truth

**The Three Cash Flow Metrics Tell a Story:**

1. **Operating Cash Flow: ₹-121 Cr**
   - Core business operations consumed cash this quarter
   - Cannot self-fund even without growth investments

2. **Free Cash Flow: ₹-443 Cr**
   - After necessary CapEx (₹-322 Cr)
   - Business burns ₹443 Cr quarterly even maintaining current operations

3. **Change in Cash: ₹-2,217 Cr**
   - **This is the alarming number**
   - Cash reserves dropped by ₹2,217 Cr in one quarter
   - Includes working capital changes, investments, and other outflows

**What This Means:**
The company isn't just unprofitable on paper - it's actively consuming cash reserves at an accelerating rate. This is not sustainable beyond 2-3 years without either:
- Dramatic operational improvement
- Additional capital raise (further shareholder dilution)
- Strategic restructuring

---

### Market Perception vs. Fundamentals

**Stock Performance:**
- 1-year return: +50% (₹895 → ₹1,343)
- Volatility: 37.22% (moderate for growth stock)

**Interpretation:**
The market has shown some optimism (50% gain), but moderate-to-high volatility (37%) indicates investors remain divided on the turnaround story. This is neither a vote of confidence nor a panic sell-off - it's uncertainty.

**Compare to:**
- Blue-chip stocks: 20-30% volatility
- Distressed companies: 60%+ volatility
- **Paytm at 37%:** Market sees risk but not disaster

**The contradiction:**
Stock up 50%, yet fundamentals show persistent losses. This suggests:
1. Market is pricing in a future turnaround (speculative)
2. Recovery from post-RBI ban lows (technical rebound)
3. Hope, not evidence, driving valuations

---

### Competitive Context

**The 14-Year Question:**
Paytm has operated for over a decade. At this maturity level:
- PhonePe: Reportedly approaching profitability
- Google Pay: Backed by profitable parent (can afford losses)
- **Paytm:** Still searching for sustainable model

**This raises the critical question:** Is this a temporary investment phase (like Amazon's early years) or a fundamentally flawed business model?

**Evidence suggests the latter:**
- 14 years is beyond "growth investment" phase
- Competitors with similar models are profitable
- Operating margin of -26.75% indicates structural, not cyclical, issues

---

### Investment/Lending Decision

**Would I lend ₹100 Crores to Paytm?**

**NO.**

**Reasoning:**
While technically they have assets to cover repayment (strong balance sheet), the negative cash flow means:
- They'd be borrowing to fund operating losses, not growth
- Repayment would come from equity raises or asset sales, not operations
- Post-RBI ban adds regulatory risk premium
- Any competent credit committee would reject this (hence their near-zero debt)

---

**Would I invest in Paytm stock?**

**ONLY AS HIGH-RISK SPECULATION** (max 5% of portfolio)

**Bull Case:**
- Management executes turnaround in next 2-3 years
- Margins improve as scale increases
- RBI ban impact fades
- Stock currently 40% below IPO (potential upside if recovery happens)

**Bear Case:**
- Business model fundamentally broken
- Competition intensifies (PhonePe, Google Pay)
- Cash burn continues
- Requires dilutive capital raise
- Eventual "zombie company" status or fire sale

**This is a binary bet:** Either they fix the model (stock could double) or they don't (stock could halve). Not suitable for risk-averse investors.

---

### Most Surprising Finding

**The IPO Cash Masquerade:**

What shocked me most was the disconnect between appearance and reality:

- **Balance sheet says:** "Strong, liquid, low-risk company"
- **Operations say:** "14-year-old business that cannot generate positive cash flow"

The ₹18,300 Cr IPO is functioning as a very expensive runway, not growth capital. Instead of accelerating a profitable model, it's funding an unprofitable one.

**Translation:** Shareholders paid ₹2,150/share in 2021 to buy time for management to figure out profitability - and with the stock at ₹1,343, that time is being purchased at shareholder expense.

---


## Methodology

**Data Source:** Yahoo Finance via yfinance library  
**Analysis Period:** Latest available quarterly data  
**Metrics Analyzed:** 
- Debt-to-Equity Ratio
- Return on Equity (ROE)
- Current Ratio
- Stock Volatility (1 year)

**Limitation**
-  Analysis based on publicly reported financials
- Does not include qualitative factors (management quality, regulatory risk)
- Point-in-time snapshot, not trending analysis

**Tools Used**
- Python
- yfinance library
- jupyter notebook

### About This Analysis

This analysis was conducted as part of my data analytics portfolio to demonstrate:
- Financial statement analysis skills
- Python for financial data extraction
- Critical thinking and business judgment
- Technical communication (code + insights)

**Connect with me:** https://www.linkedin.com/in/swetaranii/
